src/value.zig

```diff
@@ -2386,6 +2414,74 @@ pub const Value = extern union {
         };
     }
 
+    pub fn isVectorlike(val: Value) bool {
+        return switch (val.tag()) {
+            .aggregate, .repeated => true,
+
+            // Hack to auto-vectorize without needing to allocate a new `.repeated`
+            .zero,
+            .one,
+            .bool_true,
+            .bool_false,
+            .int_i64,
+            => true,
+
+            else => false,
+        };
+    }
+
+    /// Returns vectorlike length if values are compatible vectorlikes.
+    pub fn compatibleVectorlikes(values: anytype) ?usize {
+        var has_real_vector = false;
+        var length: usize = 0;
+        comptime var i: usize = 0;
+        inline while (i < values.len) : (i += 1) {
+            const val = values[i];
+            if (@TypeOf(val) == Type) {
+                has_real_vector = true;
+                length = val.vectorLen();
+            } else {
+                switch (val.tag()) {
+                    .aggregate => {
+                        const new_len = val.castTag(.aggregate).?.data.len;
+                        if (has_real_vector and length != new_len) return null;
+                        has_real_vector = true;
+                        length = new_len;
+                    },
+                    .repeated => {},
+
+                    // Hack to auto-vectorize constants without needing to allocate a new `.repeated`
+                    .zero,
+                    .one,
+                    .bool_true,
+                    .bool_false,
+                    .int_i64,
+                    => {},
+
+                    else => return null,
+                }
+            }
+        }
+        return if (has_real_vector and length != 0) length else null;
+    }
+
+    pub fn indexVectorlike(val: Value, index: usize) Value {
```

Comment:
What does this do? This looks sus

Reply:
Which part, the return or `indexVectorlike`?

Regarding the return value, that's actually unnecessary now. Originally, `compatibleVectorlikes` had slightly different length logic & this is an artifact of that. As stated above, I'll be making the non-scalarized `Type` a required argument for each `Value` function to avoid needing to do any wacky "unsound" stuff here.

Regarding `indexVectorlike`, this is just a wrapper to avoid putting logic to handle `repeated` and implicitly vectorizable values (like `zero` and `one`) in every single vectorized function.

Reply:
Apologies for the misclick- the comment was supposed to be only on the function signature.

If this function survives the other changes you are making, would you mind describing clearly what this function does in the doc comments?

Reply:
I just realized that I never responded to this comment. This is actually the only surviving function and I've added a descriptive doc comment to it.